odoo/api.py

```diff
@@ -541,35 +542,51 @@ def user(self):
 
     @lazy_property
     def company(self):
-        """ return the company in which the user is logged in (as an instance) """
-        company_ids = self.context.get('allowed_company_ids', False)
+        """Return the current company (as an instance).
+
+        If not specified in the context ('allowed_company_ids'),
+            fallback on current user main company.
+        :raise AccessError: invalid or unauthorized 'allowed_company_ids' context key content.
+        :return: current company (default=`self.user.company_id`)
+        :rtype: res.company
+        """
+        company_ids = self.context.get('allowed_company_ids', [])
         if company_ids:
-            company_id = int(company_ids[0])
-            if company_id in self.user.company_ids.ids:
-                return self['res.company'].browse(company_id)
+            if not self.su:
```

Comment:
Is this worth a comment to remind us of the rationale for special-casing `su` mode?
The commit message for that change deserves a bit more than a one-liner too :p

Reply:
Information added in method docstring.